changelogs/unreleased/gh-6726-fix-assertion-during-allocation-in-buffer-module.md

```diff
@@ -0,0 +1,4 @@
+## bugfix/small
+
+* Fix assertion when allocating a large amount of
+  memory in 'buffer' module (gh-4669).
```

Comment:
A change log entry isn't required in this case, because the "bug" could only manifest itself on debug builds so no users are affected. Please remove.

Also, please update the submodule commit SHA to point to the master.

Reply:
Remove change log and update submodule